Tests/Mocks/StubCorrelationIDGenerator.swift

```diff
@@ -1,12 +1,15 @@
 @testable import Frames
 
-final class StubCorrelationIDGenerator: CorrelationIDGenerating {
+final class StubCorrelationIDManager: CorrelationIDManaging {
     
     var generateCorrelationIDReturnValue: String!
     
     func generateCorrelationID() -> String {
         
         return generateCorrelationIDReturnValue
     }
-    
+
+    func destroyCorrelationID() {
```

Comment:
Any ideas how to use this in unit tests? currently destroyCorrelationID gets only called within the success of token generation so stub object wont even reach that point.

Reply:
The stub does get called as part of `CheckoutAPIClientTests`. Anywhere where `stubRequestExecutor.executeCalledWithCompletionHandler` is called, the completion block with destroy is called. You should be able to add this check into one or all of the existing tests that call this.

Reply:
I think the test in `StubCorrelationIDGeneratorTests` is fine, as you test that it regenerates a new one.

Here you'd add `var destroyCorrelationIDCalled = false` and set it to `true` in this method. Then you'd test that `destroyCorrelationIDCalled` is `true` in the dependant class.

Reply:
Updated.